charmcraft/parts.py

```diff
@@ -99,7 +101,11 @@ def get_build_commands(self) -> List[str]:
         """Return a list of commands to run during the build step."""
         options = cast(CharmPluginProperties, self._options)
 
-        build_env = dict(LANG="C.UTF-8", LC_ALL="C.UTF-8")
+        build_env = dict(
+            LANG="C.UTF-8",
+            LC_ALL="C.UTF-8",
+            PYTHONUSERBASE=self._part_info.part_build_dir / charm_builder.STAGING_VENV_DIRNAME,
```

Comment:
I don't see why this is needed considering that this is determined by running the interpreter in the venv and everything else is determined from that symlink

Reply:
It's only being used is to recover the `lib/pythonX.Y/site-packages` directory name, so I think we can drop it and use `site.USER_SITE` relative to `site.USER_BASE`?

Reply:
Dropped `PYTHONUSERBASE` and `site.USER_SITE`, now obtaining the path to `site-packages` based on version information.